agent/http.go

```diff
@@ -359,8 +359,9 @@ func parseWait(resp http.ResponseWriter, req *http.Request, b *structs.QueryOpti
 
 // parseConsistency is used to parse the ?stale and ?consistent query params.
 // Returns true on error
-func parseConsistency(resp http.ResponseWriter, req *http.Request, b *structs.QueryOptions) bool {
+func (s *HTTPServer) parseConsistency(resp http.ResponseWriter, req *http.Request, b *structs.QueryOptions) bool {
 	query := req.URL.Query()
+	b.AllowStale = !b.RequireConsistent && s.agent.config.HttpApiConfig.AllowStale
```

Comment:
This doesn't look right since `b.RequireConsistent` hasn't been set at this point. I think you can write this better as

```go
if b.AllowStale && b.RequireConsistent { error 400 }
if !b.AllowStale && !b.RequireConsistent {
    b.AllowStale = s.agent.config.HTTPConfig.AllowStale
}
```